modules/packages/RangeChunk.chpl

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2004-2016 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ * 
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * 
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module RangeChunk {
+  enum RemElems {
+    Thru,
+    Pack,
+    Mod
+  }
+  use RemElems;
+  use BoundedRangeType;
+
+  /*
+   * Range-based interface, iterator and query.
+   * idx in query is zero based.
+   */
+
+  iter chunks(
+    r: range(?RT, bounded, ?S),
```

Comment:
All of the function arguments in this module break the conventions we've used throughout the rest of the modules, and it makes it hard to read in my opinion.

If the arguments all fit on a single 80 character line I'd prefer that.  If they don't, I'd like them indented to one space past the level of the opening parenthesis instead of to the same level as the code inside the function body.